examples/ex23.cpp

```diff
@@ -0,0 +1,988 @@
+void detJ_inv_JT_J_abs(const Vector &x, CartesianPML * pml, DenseMatrix &M);
+
+Array2D<double> comp_domain_bdr;
+Array2D<double> domain_bdr;
+
+double mu = 1.0;
+double epsilon = 1.0;
+double omega;
+int dim;
+bool exact_known = false;
+
+enum prob_type
+{
+   beam,     // PML on one end of the domain
+   scatter,  // Scattering in a square or a cube
+   lshape,   // Scattering in 1/4 of a square
+   fichera,   // Scattering in 1/8 of a cube
+   load_src // point source with PML all around
+};
+prob_type prob;
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   const char *mesh_file = nullptr;
+   int order = 1;
+   int ref_levels = 3;
+   int iprob = 4;
+   double freq = 5.0;
+   bool herm_conv = true;
+   bool visualization = 1;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree).");
+   args.AddOption(&iprob, "-prob", "--problem", "Problem case"
+                  " 0: General, 1: beam, 2: scatter, 3: lshape, 4: fichera");
+   args.AddOption(&ref_levels, "-ref", "--refinements",
+                  "Number of refinements");
+   args.AddOption(&mu, "-mu", "--permeability",
+                  "Permeability of free space (or 1/(spring constant)).");
+   args.AddOption(&epsilon, "-eps", "--permittivity",
+                  "Permittivity of free space (or mass constant).");
+   args.AddOption(&freq, "-f", "--frequency",
+                  "Frequency (in Hz).");
+   args.AddOption(&herm_conv, "-herm", "--hermitian", "-no-herm",
+                  "--no-hermitian", "Use convention for Hermitian operators.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+
+   // 2. Setup the (serial) mesh on all processors.
+   if (iprob > 4) { iprob = 4; }
+   prob = (prob_type)iprob;
+
+   if (!mesh_file)
+   {
+      exact_known = true;
+      switch (prob)
+      {
+         case beam:
+            mesh_file = "../data/beam-hex.mesh";
+            break;
+         case scatter:
+            mesh_file = "../data/square_w_hole.mesh";
+            break;
+         case lshape:
+            mesh_file = "../data/l-shape.mesh";
+            break;
+         case fichera:
+            mesh_file = "../data/fichera.mesh";
+            break;
+         default:
+            exact_known = false;
+            mesh_file = "../data/inline-quad.mesh";
+            break;
+      }
+   }
+
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   Mesh * mesh = new Mesh(mesh_file, 1, 1);
+   dim = mesh->Dimension();
+
+   //Angular frequency
+   omega = 2.0 * M_PI * freq;
+
+   // Setup PML length
+   Array2D<double> lngth(dim, 2);  lngth = 0.0;
```

Comment:
Why not name this variable "length" for better readability?

Reply:
renamed